src/Bicep.Core/Analyzers/Linter/DependsOnRemovalRewriter.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.Core.Visitors;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    // Looks for resources where a dependency can already be inferred by the structure of the resource declaration.
+    // 
+    // As an example, because the below resource already has a reference to 'otherRes' in the name property, the dependsOn is not adding anything:
+    //   resource myRes 'My.Rp/myResource@2020-01-01' = {
+    //     name: otherRes.name
+    //     dependsOn: [
+    //       otherRes
+    //     ]
+    //   }
+    public class DependsOnRemovalRewriter : SyntaxRewriteVisitor
```

Comment:
>DependsOnRemovalRewriter [](http://example.com/codeflow?start=17&length=24)

Is this the same rewriter that we have in the decompiler? If yes, should we unify them?

Reply:
It is and I have moved it to Bicep.Core to eliminate circular dependency.